### ISSUE TITLE
_abc_registry errors with py3.8

### ISSUE THREAD

**Issue by awais786** [python/typing#721]
```
    from typing import Any
  File "/.tox/django22/lib/python3.8/site-packages/typing.py", line 1357, in <module>
    class Callable(extra=collections_abc.Callable, metaclass=CallableMeta):
  File "/.tox/django22/lib/python3.8/site-packages/typing.py", line 1005, in __new__
    self._abc_registry = extra._abc_registry
AttributeError: type object 'Callable' has no attribute '_abc_registry'
```

**Comment by gvanrossum** [comment]
Hm, with Python 3.8, typing.py should not come from site-packages. This seems to point to some installation or configuration error on your part. Please seek help elsewhere.

**Comment by simkessy** [comment]
I'm on python 3.7 and latest pip and recently started getting this error. My solution was to remove typings. Is there a better solution?

**Comment by gvanrossum** [comment]
The solution is probably `pip uninstall typing`.

**Comment by simkessy** [comment]
This worked. Any idea why this might happen all of a sudden?

**Comment by gvanrossum** [comment]
Maybe you or a script incorrectly ran 'pip install typing'. Or maybe some other thing that you installed had an incorrect dependency on typing. It would require a lot of research to find out what happened, and it would have to be done on your computer. I haven't heard of this before (other than the initial report in this issue).

**Comment by gvanrossum** [comment]
I wonder if there's a way to make this package refuse to install itself on any Python version that has a typing module in the stdlib. (I think starting with 3.5.)

**Comment by ethanhs** [comment]
The `python_requires` for typing https://github.com/python/typing/blob/08a537d484edddb563288982f8073d241fec5535/setup.py#L70 allows it to be installed on 3.4+

It was changed in https://github.com/python/typing/pull/722, and it seems you proposed disallowing installation on 3.5+. Seeing as all versions of Python 3 that don't have typing in the stdlib are not longer supported themselves, I guess the pypi package only has use for 2.7?

**Comment by JukkaL** [comment]
> Seeing as all versions of Python 3 that don't have typing in the stdlib are not longer supported themselves, I guess the pypi package only has use for 2.7?

2.7 is also unsupported now, though clearly it's still being used a lot. Some projects also decide to continue to support unsupported Python 3 versions.  For a very widely used package such as `typing` I'd argue that it makes sense to continue supporting older Python releases after their end of life, at least unless it's very difficult.

**Comment by gvanrossum** [comment]
Maybe the right thing to do is just not to do any new releases of the `typing` package itself, but focus our attention on `typing_extensions` (also in this repo)? If `typing` works now for Python releases that are no longer supported, it should just keep working, right? Or perhaps we could do one final release, which refuses to let itself be installed in 3.5 and up to prevent the original problem of this issue.

**Comment by JelleZijlstra** [comment]
> 2.7 is also unsupported now

Is that true for typing tools in general? I believe typeshed for example should still accept Python 2.7 fixes.

**Comment by JukkaL** [comment]
Yeah, typeshed still supports 2.7. I meant that CPython 2.7 has reached it's End of Life.

(Tangentially related: even if typeshed doesn't support older 3.x releases officially, more recent 3.x stubs generally work just fine with older Python 3 releases.)

**Comment by basnijholt** [comment]
Would releasing a new version of `typing` fix https://github.com/pypa/pip/issues/8272 now that https://github.com/python/typing/pull/722 is merged?

**Comment by gvanrossum** [comment]
I could do that, if @JukkaL and/or @ilevkivskyi agree that now's a good time.

**Comment by ilevkivskyi** [comment]
Yes, thanks, this is a good time.

**Comment by maxwellmckinnon** [comment]
> I wonder if there's a way to make this package refuse to install itself on any Python version that has a typing module in the stdlib. (I think starting with 3.5.)

This is the right answer

**Comment by gvanrossum** [comment]
@maxwellmckinnon I'm still a wheels newbie. Do you know how to do that from a wheel?

**Comment by gvanrossum** [comment]
Never mind the answer is a [few comments up](https://github.com/python/typing/issues/721#issuecomment-631907293).

**Comment by gvanrossum** [comment]
Okay, I've done a release of typing 3.7.4.2. I changed python_requirements so that it *only* installs on 2.7 and 3.4. I couldn't find an installable binary of 3.4 (it's well out of support) so I only tested on 2.7 -- if anyone is still using 3.4 and finds a problem they know where to find us. For this reason there also isn't a wheel for 3.4, only an sdist. The changes I made to setup.py are now in master and tagged "3.7.4.2".
